[EuiIcon] Add move glyph

andreadelrio opened the pull request:
## Summary

This new icon is needed for an upcoming [new feature](https://github.com/elastic/kibana/issues/151233) in Dashboard where we add hover actions to panels. One of these actions will be "move panel". 

<img width="934" alt="image" src="https://github.com/elastic/eui/assets/4016496/c234891f-54e4-42ef-8e86-fe8fc9a31653">


<img width="324" alt="image" src="https://github.com/elastic/eui/assets/4016496/6b4a9b69-16d4-4225-bbc3-5bc05842117f">


## QA

Remove or strikethrough items that do not apply to your PR.

### General checklist

- Browser QA
    - [x] Checked in both **light and dark** modes
    - [x] Checked in **mobile**
    - [x] Checked in **Chrome**, **Safari**, **Edge**, and **Firefox**
    - [x] Checked for **accessibility** including keyboard-only and screenreader modes
- Docs site QA
    - [x] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting)**
    - [x] Props have proper **autodocs** (using `@default` if default values are missing) and **[playground toggles](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/playgrounds.md)**
    - [x] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples
- Code quality checklist
    - [x] Added or updated **[jest](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/unit-testing.md) and [cypress](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/cypress-testing.md) tests**
- Release checklist
    - [x] A **[changelog](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/changelogs.md)** entry exists and is marked appropriately.
   - [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)
- Designer checklist 
    - [ ] Updated the **[Figma](https://www.figma.com/community/file/964536385682658129)** library counterpart

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_7789/
- Storybook: https://eui.elastic.co/pr_7789/storybook

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/2033)
* Commit: 8b30c1d539255664672f4c3e073bc6c0da2c05fc

### Failed CI Steps
* [:hammer: EUI pull request test](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/2033#018fa8bb-33cf-48ce-a5cf-dc0dbeff1674)

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/2033","number":2033,"commit":"8b30c1d539255664672f4c3e073bc6c0da2c05fc"}],"number":2033}
buildkite-pr-comment-->

mgadewoll commented:
Hei @andreadelrio thanks for this PR! There seems to an issue with missing files (added svg, generated icon component) which looks to be a small issue introduced when we updated to a monorepo in this [PR](https://github.com/elastic/eui/pull/7736).
I updated your PR to [fix it](https://github.com/elastic/eui/pull/7789/commits/9a6698194687e9470c97b337a95c14db10671dc7) and [add the missing files](https://github.com/elastic/eui/pull/7789/commits/d4d485e197beedd38fe960bf29ab13addc6c5e33). Hope you don't mind 🙂

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_7789/
- Storybook: https://eui.elastic.co/pr_7789/storybook

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/2041)
* Commit: 2bf2815692383078077193e89aa4c0786d702b94

### History
* :broken_heart: [Build #2033](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/2033) failed 8b30c1d539255664672f4c3e073bc6c0da2c05fc

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/2041","number":2041,"commit":"2bf2815692383078077193e89aa4c0786d702b94"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/2033","number":2033,"commit":"8b30c1d539255664672f4c3e073bc6c0da2c05fc"}],"number":2041}
buildkite-pr-comment-->

andreadelrio commented:
> Hei @andreadelrio thanks for this PR! There seems to an issue with missing files (added svg, generated icon component) which looks to be a small issue introduced when we updated to a monorepo in this [PR](https://github.com/elastic/eui/pull/7736). I updated your PR to [fix it](https://github.com/elastic/eui/pull/7789/commits/9a6698194687e9470c97b337a95c14db10671dc7) and [add the missing files](https://github.com/elastic/eui/pull/7789/commits/d4d485e197beedd38fe960bf29ab13addc6c5e33). Hope you don't mind 🙂

Thanks Lene!

cee-chen commented:
@MichaelMarcialis Ping for designer review on this - or should we just go ahead and merge it as-is?

MichaelMarcialis reviewed (approved):
Thanks for the ping. This icon looks nice, @andreadelrio. The only question I have is whether it makes sense for the intersecting lines to have a 2px stroke width (as they do currently), or if it would be more consistent for them to be 1px. Most (but not all) of our arrow-based icons tend to use a 1px stroke. However, if it was better for alignment or shrinking of the icon to deviate, I'm also not opposed to it. Just curious.

That said, approving now to not hold you up, assuming you had valid reasons for the line thickness.

andreadelrio commented:
> Thanks for the ping. This icon looks nice, @andreadelrio. The only question I have is whether it makes sense for the intersecting lines to have a 2px stroke width (as they do currently), or if it would be more consistent for them to be 1px. Most (but not all) of our arrow-based icons tend to use a 1px stroke. However, if it was better for alignment or shrinking of the icon to deviate, I'm also not opposed to it. Just curious.
> 
> That said, approving now to not hold you up, assuming you had valid reasons for the line thickness.

@MichaelMarcialis Thanks for the review. The reasoning is that I'm trying to mimic the native `move` cursor that web browsers have and I feel like the 2px stroke width achieves a better result.
